sdk/core/azure-core/src/test/java/com/azure/core/models/CloudEventTests.java

```diff
@@ -0,0 +1,309 @@
+        String jsonData = getTestPayloadFromFile("CloudEventBinaryData.json");
+
+        byte[] data = "c2FtcGxlYmluYXJ5ZGF0YQ==".getBytes(StandardCharsets.UTF_8);
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        byte[] eventData = events.get(0).getData().toBytes();
+
+        assertNotNull(eventData);
+
+        assertArrayEquals(data, eventData);
+    }
+
+    @Test
+    public void consumeCloudEventJsonData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventJsonData.json");
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        ContosoItemReceivedEventData data = events.get(0).getData().toObject(ContosoItemReceivedEventData.class,
+            SERIALIZER);
+        assertNotNull(data);
+
+        assertEquals("512d38b6-c7b8-40c8-89fe-f46f9e9622b6", data.getItemSku());
+
+        Map<String, Object> additionalProperties = events.get(0).getExtensionAttributes();
+
+        assertNotNull(additionalProperties);
+        assertTrue(additionalProperties.containsKey("foo"));
+        assertEquals("bar", additionalProperties.get("foo"));
+    }
+
+    @Test
+    public void consumeCloudEventXmlData() throws IOException {
+        String jsonData = getTestPayloadFromFile("CloudEventXmlData.json");
+
+        List<CloudEvent> events = CloudEvent.fromString(jsonData);
+
+        assertNotNull(events);
+        assertEquals(1, events.size());
+
+        assertEquals(events.get(0).getExtensionAttributes().get("comexampleothervalue"), 5);
+
+        String xmlData = events.get(0).getData().toString();
+
+        assertEquals("<much wow=\"xml\"/>", xmlData);
+    }
+
+    @Test
+    public void testSerializeByteDataAndAllAttributes() {
+        BinaryData binaryData = BinaryData.fromBytes("AAA".getBytes(StandardCharsets.UTF_8));
+        CloudEvent cloudEvent = new CloudEvent("/testSource", "CloudEvent.Test", binaryData, CloudEventDataFormat.BYTES, "bytes")
+            .setDataSchema("/testSchema")
+            .setSubject("testSubject")
+            .setTime(OffsetDateTime.now())
+            .setSpecVersion("1.0")
+            .addExtensionAttribute("foo", "value");
+        try {
+            String serializedString = SERIALIZER.getJacksonAdapter().serialize(cloudEvent, SerializerEncoding.JSON);
+            CloudEvent deserializedCloudEvent = CloudEvent.fromString(serializedString).get(0);
+            assertEquals(cloudEvent.getData().toString(), deserializedCloudEvent.getData().toString());
+            assertEquals(new String(Base64.getDecoder().decode(cloudEvent.getData().toBytes())), "AAA");
+            compareCloudEventContent(cloudEvent, deserializedCloudEvent);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
wouldn't you want to fail your test ? instead of throwing RuntimeException 
Same in other tests.